### PR TITLE
Update to idiomatic Spring Boot + Kotlin application

### PR DESCRIPTION
**Pull request by sdeleuze**
I have also submitted a pull request on kotlin-examples with these changes: https://github.com/JetBrains/kotlin-examples/pull/49

### CODE REVIEW

File: pages/docs/tutorials/spring-boot-restful.md [hunk 1]
```diff
@@ -19,26 +19,26 @@ Note that all classes in this tutorial are in the `org.jetbrains.kotlin.demo` pa
...
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version" // Required for Kotlin integration
+	classpath "org.jetbrains.kotlin:kotlin-allopen:$kotlin_version" See https://kotlinlang.org/docs/reference/compiler-plugins.html#kotlin-spring-compiler-plugin
         classpath "org.springframework.boot:spring-boot-gradle-plugin:$spring_boot_version"
     }
 }
 
 apply plugin: 'eclipse'
-apply plugin: 'idea'
 apply plugin: 'kotlin' // Required for Kotlin integration
+apply plugin: "kotlin-spring" See https://kotlinlang.org/docs/reference/compiler-plugins.html#kotlin-spring-compiler-plugin
```
Comment (AlexCzar) [inline pages/docs/tutorials/spring-boot-restful.md]: missing `//` before comment

File: pages/docs/tutorials/spring-boot-restful.md [hunk 2]
```diff
@@ -76,70 +72,35 @@ class GreetingController {
...
 
 As can be seen, this is again pretty much a one-to-one translation of Java to Kotlin, with nothing special required for Kotlin.
 
 ### Creating the Application class
-Finally we need to define an Application class. As Spring Boot looks for a public static main method, we need to define this in Kotlin
-using the *@JvmStatic* annotation. For this, we create a standard *Application* class and define a companion object inside where we can then create
-a function annotated with *@JvmStatic*
-
-Note: JvmStatic is an annotation in Kotlin which is used for interoperability with Java, so that the resulting method is defined as static when called from Java.
+Finally we need to define an Application class. As Spring Boot looks for a public static main method, we need to define this in Kotlin. It could be done with the *@JvmStatic* annotation and a companion object but here we prefer using a [top-level functions]({{ url_for('page', page_path="docs/reference/functions") }}) defined outside Application class since it leads to more concise and clean code.
```
Comment (AlexCzar) [inline pages/docs/tutorials/spring-boot-restful.md]: should be `a [top-level-function]` - singular.

File: pages/docs/tutorials/spring-boot-restful.md [hunk 3]
```diff
@@ -19,26 +19,26 @@ Note that all classes in this tutorial are in the `org.jetbrains.kotlin.demo` pa
...
 buildscript {
     ext.kotlin_version = '{{ site.data.releases.latest.version }}' // Required for Kotlin integration
-    ext.spring_boot_version = '1.5.3.RELEASE'
+    ext.spring_boot_version = '1.5.4.RELEASE'
     repositories {
         jcenter()
     }
     dependencies {
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version" // Required for Kotlin integration
+	classpath "org.jetbrains.kotlin:kotlin-allopen:$kotlin_version" See https://kotlinlang.org/docs/reference/compiler-plugins.html#kotlin-spring-compiler-plugin
```
Comment (AlexCzar) [inline pages/docs/tutorials/spring-boot-restful.md]: missing `//` before comment

File: pages/docs/tutorials/spring-boot-restful.md [hunk 4]
```diff
@@ -19,26 +19,26 @@ Note that all classes in this tutorial are in the `org.jetbrains.kotlin.demo` pa
 ### Defining the project and dependencies
 {{ site.text_using_gradle }}
 
-The Gradle file is pretty much standard for Spring Boot. The only difference is the structure layout for source folders for Kotlin, and the required Kotlin dependencies
+The Gradle file is pretty much standard for Spring Boot. The only differences are the structure layout for source folders for Kotlin, the required Kotlin dependencies and the *kotlin-spring* Gradle plugin which prevents using *open* modifier at class and member function level for Spring beans proxified with CGLIB.
```
Comment (AlexCzar) [inline pages/docs/tutorials/spring-boot-restful.md]: As far as I understand *kotlin-spring* plugin does not _prevent_ using `open`, quite the opposite—it automatically puts it in place during compilation.

**Comment by sdeleuze** [comment]
@AlexCzar Thanks for your feedback, I am off today but I will update the PR monday first thing.

**Comment by hhariri** [comment]
@sdeleuze once done we can merge, thanks.

**Comment by sdeleuze** [comment]
@AlexCzar I have taken you feedbacks in account.

**Comment by sdeleuze** [comment]
I have updated the commit in order to be consistant with my latest https://github.com/JetBrains/kotlin-examples/pull/49 update.

**Comment by yole** [comment]
Thanks! I'll push a couple minor changes on top of that.

**Pull request merged**